pkg/minishift/config/validations.go

```diff
@@ -111,14 +115,44 @@ func IsValidProxy(name string, uri string) error {
 	return nil
 }
 
-func IsValidUrl(_ string, isoURL string) error {
-	if isoURL == minishiftConstants.B2dIsoAlias || isoURL == minishiftConstants.CentOsIsoAlias || isoURL == minishiftConstants.MinikubeIsoAlias {
+func IsValidISOUrl(_ string, isoURL string) error {
+	for _, isoAlias := range minishiftConstants.ValidIsoAliases {
+		if isoURL == isoAlias {
+			return nil
+		}
+	}
+	if !strings.HasSuffix(isoURL, ".iso") {
+		return fmt.Errorf("'%s' url is not valid", isoURL)
+	}
+
+	match, _ := regexp.MatchString(`^https?://`, isoURL)
+	if match {
 		return nil
 	}
+
+	if runtime.GOOS == "windows" {
+		match, _ := regexp.MatchString("^file://[a-zA-Z]:/.+", isoURL)
+		if !match {
+			return fmt.Errorf("'%s' url is not valid", isoURL)
+		}
+		if filehelper.Exists(strings.Replace(strings.TrimPrefix(isoURL, "file://"), "/", "\\", -1)) {
+			return nil
+		}
+	} else {
+		match, _ := regexp.MatchString("^file:///.+", isoURL)
+		if !match {
+			return fmt.Errorf("'%s' url is not valid", isoURL)
+		}
+		if filehelper.Exists(strings.TrimPrefix(isoURL, "file://")) {
+			return nil
+		}
+	}
+
 	_, err := url.ParseRequestURI(isoURL)
```

Comment:
Can we put this check right after we do the http check, that way we can fail earlier and don't have to go through the relatively complicated check for the `file:///` thing.